### PR TITLE
refactor: Migrate GroupCreationViewModel to Typescript

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

File: src/script/view_model/content/GroupCreationViewModel.ts [hunk 1]
```diff
@@ -19,18 +19,43 @@
...
+import {UserRepository} from '../../user/UserRepository';
+import {Conversation} from '../../entity/Conversation';
+import {User} from '../../entity/User';
 
 export class GroupCreationViewModel {
+  isTeam: ko.PureComputed<boolean>;
+  isShown: ko.Observable<boolean>;
+  state: ko.Observable<string>;
+  private isCreatingConversation: boolean;
+  private method: string;
```
Comment (bennycode) [inline src/script/view_model/content/GroupCreationViewModel.ts]: Can we find a better name here?

File: src/script/view_model/content/GroupCreationViewModel.ts [hunk 2]
```diff
@@ -208,16 +235,20 @@ export class GroupCreationViewModel {
...
     this._trackAddParticipants(conversationEntity);
-  }
+  };
 
-  _trackGroupCreationSucceeded(conversationEntity) {
-    const attributes = {
+  _trackGroupCreationSucceeded = (conversationEntity: Conversation): void => {
+    const attributes: {
+      is_allow_guests?: boolean;
+      method: string;
```
Comment (bennycode) [inline src/script/view_model/content/GroupCreationViewModel.ts]: Let's use a type like:
```ts
export GroupCreationSource = 'start_ui' | 'conversation_details';
```

Comment (thisisamir98) [reply src/script/view_model/content/GroupCreationViewModel.ts]: done

File: src/script/view_model/content/GroupCreationViewModel.ts (hunk 1 above)
Comment (thisisamir98) [reply src/script/view_model/content/GroupCreationViewModel.ts]: done

**Pull request merged**